mozetl/topline/topline_dashboard.py

```diff
@@ -122,16 +122,17 @@ def write_dashboard_data(df, bucket, prefix, mode):
 
     # create a temporary directory to dump files into
     path = tempfile.mkdtemp()
-
-    # write dataframe to working directory
-    df.repartition(1).write.csv(path, header=True, mode="overwrite")
-
-    # find the file that was created
-    filepath = None
-    for name in os.listdir(path):
-        if name.endswith(".csv"):
-            filepath = os.path.join(path, name)
-            break
+    filepath = os.path.join(path, 'temp.csv')
+
+    with open(filepath, 'wb') as f:
+        f.write(','.join(df.columns) + '\n')
+        rows = df.collect()
+        for row in rows:
+            try:
+                f.write(','.join([unicode(c) for c in row]))
+                f.write('\n')
+            except UnicodeEncodeError:
+                logger.exception("exception while processing csv")
```

Comment:
Ok, can you confirm that writing bare 'strings' (rather than u'unicode' strings) works ok in combination with unicode field values?

It might be worthwhile to factor out some "write this dataframe to csv" functionality and add some tests. handling all the unicode/utf8 corner cases in python2 always seems to be fraught with peril.